pkg/cmd/admin/policy/whatcanido.go

```diff
@@ -16,32 +20,55 @@ import (
 	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
 )
 
-const WhatCanIDoRecommendedName = "what-can-i-do"
+const WhatCanIDoRecommendedName = "can-i"
 
 type whatCanIDoOptions struct {
-	namespace string
-	client    client.SelfSubjectRulesReviewsNamespacer
-
-	out io.Writer
+	AllNamespaces     bool
+	ListAll           bool
+	Quiet             bool
+	Namespace         string
+	RulesReviewClient client.SelfSubjectRulesReviewsNamespacer
+	SARClient         client.SubjectAccessReviews
+
+	Verb         string
+	Resource     unversioned.GroupVersionResource
+	ResourceName string
+
+	Out io.Writer
 }
 
 // NewCmdWhatCanIDo implements the OpenShift cli who-can command
 func NewCmdWhatCanIDo(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
-	options := &whatCanIDoOptions{out: out}
+	o := &whatCanIDoOptions{
+		Out: out,
+	}
 
 	cmd := &cobra.Command{
-		Use:   name,
-		Short: "List what I can do in this namespace",
-		Long:  "List what I can do in this namespace",
+		Use:   name + " VERB RESOURCE [NAME]",
+		Short: "Test whether an action is allowed",
+		Long:  "Test whether an action is allowed",
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := options.complete(f, args); err != nil {
+			if reflect.DeepEqual(args, []string{"educate", "dolphins"}) {
+				fmt.Fprintln(o.Out, "Only liggitt can educate dolphins.")
+				return
+			}
+
+			if err := o.Complete(f, args); err != nil {
 				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
 			}
 
-			kcmdutil.CheckErr(options.run())
+			allowed, err := o.Run()
+			kcmdutil.CheckErr(err)
+			if !allowed {
+				os.Exit(2)
+			}
 		},
 	}
 
+	cmd.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", o.AllNamespaces, "Test the specified action in all namespaces.")
```

Comment:
"Check whether the specified action is allowed in all namespaces you have access to"